skimage/util/_backends.py

```diff
@@ -0,0 +1,153 @@
+
+    for backend in backends_:
+        backends[backend.name] = {"implementation": backend}
+        try:
+            info = backend_infos[backend.name]
+            # Double () to load and then call the backend information function
+            backends[backend.name]["info"] = info.load()()
+        except KeyError:
+            pass
+
+    return backends
+
+
+def dispatchable(func):
+    """Mark a function as dispatchable.
+
+    When a decorated function is called the installed backends are
+    searched for an implementation. If no backend implements the function
+    then the scikit-image implementation is used.
+    """
+    func_name = func.__name__
+    func_module = get_module_name(func)
+
+    # If no backends are installed at all or dispatching is disabled,
+    # return the original function. This way people who don't care about it
+    # don't see anything related to dispatching
+    if disable_dispatching() or not all_backends():
+        return func
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        # Backends are tried in alphabetical order, this makes things
+        # predictable and stable across runs. Might need a better solution
+        # when it becomes common that users have more than one backend
+        # that would accept a call.
+        for name in sorted(all_backends()):
+            backend = all_backends()[name]
+            # If we have a BackendInformation object we check if the
+            # function we are looking for is implemented in the backend
+            if "info" in backend:
+                if (
+                    f"{func_module}:{func_name}"
+                    not in backend["info"].supported_functions
+                ):
+                    continue
+
+            backend_impl = backend["implementation"].load()
+
+            # Allow the backend to accept/reject a call based on the function
+            # name and the values of the arguments
+            wants_it = backend_impl.can_has(
+                f"{func_module}:{func_name}", *args, **kwargs
+            )
+            if not wants_it:
+                continue
+
+            # can_has("foo", ...) might be True, but the function might not actually
+            # be implemented in the backend
+            func_impl = backend_impl.get_implementation(f"{func_module}:{func_name}")
+            if func_impl is not None:
+                warnings.warn(
+                    f"Call to '{func_module}:{func_name}' was dispatched to"
+                    f" the '{name}' backend. Set SKIMAGE_NO_DISPATCHING=1 to"
+                    " disable this.",
+                    DispatchNotification,
+                    # XXX from where should this warning originate?
+                    # XXX from where the function that was dispatched was called?
+                    # XXX or from where the user called a function that called
+                    # XXX a function that was dispatched?
```

Comment:
This warning should probably be raised only once, when valid backends are detected, perhaps in `all_backends`? To make dispatching of a specific call transparent I'd probably start with logging instead?